src/libsyntax/parse/parser.rs

```diff
@@ -946,6 +994,10 @@ impl<'a> Parser<'a> {
     pub fn span_fatal(&self, sp: Span, m: &str) -> DiagnosticBuilder<'a> {
         self.sess.span_diagnostic.struct_span_fatal(sp, m)
     }
+    pub fn span_fatal_err(&self, sp: Span, err: Errors) -> DiagnosticBuilder<'a> {
+        let res = err.span_err(sp, self.diagnostic());
+        res
```

Comment:
It allows to reduce code and improve clarity so I definitely think we should keep it (but maybe not as public?).